hpx/runtime/serialization/detail/polymorphic_nonintrusive_factory.hpp

```diff
@@ -189,15 +193,47 @@ namespace hpx { namespace serialization { namespace detail
 #define HPX_SERIALIZATION_REGISTER_CLASS_NAME(Class, Name)                    \
     namespace hpx { namespace serialization { namespace detail {              \
         template <> HPX_ALWAYS_EXPORT                                         \
-        char const* get_serialization_name<Class>()                           \
+        struct get_serialization_name<Class>                                  \
         {                                                                     \
-            return Name;                                                      \
-        }                                                                     \
+            char const* operator()()                                          \
+            {                                                                 \
+                return Name;                                                  \
+            }                                                                 \
+        };                                                                    \
     }}}                                                                       \
     template hpx::serialization::detail::register_class<Class>                \
         hpx::serialization::detail::register_class<Class>::instance;          \
 /**/
+#define HPX_SERIALIZATION_REGISTER_CLASS_NAME_TEMPLATE(Parameters, Template, Name) \
+    namespace hpx { namespace serialization { namespace detail {              \
+        Parameters                                                            \
+        struct get_serialization_name<Template>                               \
+        {                                                                     \
+            char const* operator()()                                          \
+            {                                                                 \
+                return Name;                                                  \
+            }                                                                 \
+        };                                                                    \
+    }}}                                                                       \
+/**/
 #define HPX_SERIALIZATION_REGISTER_CLASS(Class)                               \
     HPX_SERIALIZATION_REGISTER_CLASS_NAME(Class, BOOST_PP_STRINGIZE(Class))   \
-
+/**/
+#define HPX_SERIALIZATION_REGISTER_CLASS_TEMPLATE(Parameters, Template)       \
```

Comment:
Unfortunately, this will break when having more than one template parameter, see here:
http://melpon.org/wandbox/permlink/QSsUVv9Q6av1XMty

An alternative would be to only pass the template arguments in the form of `(typename A, typename B)` and then expand it to `template <HPX_UTIL_STRIP(Parameters)>`